src/validator/checks.py

```diff
@@ -0,0 +1,215 @@
+import re
+
+import pandas as pd
+
+
+def column_name_check(dataframe):
+    good_column_names = {
+        'contact_name',
+        'contact_email',
+        'contact_phone',
+        'category_code',
+        'location_id',
+        'address_primary',
+        'city',
+        'state',
+        'zip_code',
+        'zip_code_suffix',
+        'unit_count',
+        'building_type_code',
+        'non_bsl_code',
+        'bsl_lacks_address_flag',
+        'latitude',
+        'longitude',
+    }
+
+    dataframe_column_names = set(dataframe.columns)
+
+    matched_names = good_column_names & dataframe_column_names
+    missing_good_names = good_column_names - dataframe_column_names
+    extra_dataframe_names = dataframe_column_names - good_column_names
+
+    return (matched_names, missing_good_names, extra_dataframe_names)
+
+
+def category_code_check(dataframe):
+
+    try:
+        dataframe['category_code'] = dataframe['category_code'].astype('int64')
+    except ValueError:
+        return None, 'Non-integer, non-numeric or empty category_code values present'
+
+    if not dataframe['category_code'].between(1, 7).all():
+        return None, 'One or more category codes were not between 1 and 7.'
+
+    return dataframe, 'Valid'
+
+
+def contact_name_check(row):
+    if not row['contact_name']:
+        return 'contact_name must include the name of the person preparing the challenge'
+    return 'Valid'
+
+
+def contact_email_check(row):
+    if not re.match(r'[^@]+@[^@]+\.[^@]+', row['contact_email']):
```

Comment:
**thought**: I wonder if something like https://pypi.org/project/email-validator/ would give a better solution.

Reply:
I thought about that, but I want to keep the dependencies as light as possible so as to not screw up someone's arcgispro-py3 environment.